src/target_intacct/hours_per_week_denominator.py

```diff
@@ -0,0 +1,156 @@
+import json
+import sys
+
+import pandas as pd
+
+import singer
+
+from .utils import format_date_to_intacct, get_input, set_journal_entry_value
+
+logger = singer.get_logger()
+
+"""
+Uploads Hours Per Week Denominator Statistical Journals to Intacct
+
+Retrieves objects from Intacct API for verifying input data
+Calls load_entries method
+Sends entries for uploading to Intacct
+"""
+
+
+def hours_per_week_denominator_upload(intacct_client, object_name) -> None:
+    logger.info("Starting upload.")
+
+    # Load Current Data in Intacct for input verification
+    employee_ids = intacct_client.get_entity(
+        object_type="employees", fields=["EMPLOYEEID"]
+    )
+    business_units = intacct_client.get_entity(
+        object_type="classes", fields=["CLASSID"]
+    )
+    location_ids = intacct_client.get_entity(
+        object_type="locations", fields=["LOCATIONID"]
+    )
+    practice_area_ids = intacct_client.get_entity(
+        object_type="departments", fields=["DEPARTMENTID"]
+    )
+
+    # Journal Entries to be uploaded
+    journal_entries = load_hours_per_week_denominator_entries(
+        employee_ids, business_units, location_ids, practice_area_ids, object_name
+    )
+
+    # Post the journal entries to Intacct
+    for entry in journal_entries:
+        intacct_client.post_journal(entry)
+
+    logger.info("Upload completed")
+
+
+"""
+Loads inputted data into Hours Per Week Denominator Statistical Journal Entries
+"""
+
+
+def load_hours_per_week_denominator_entries(
+    employee_ids, business_units, location_ids, practice_area_ids, object_name
+):
+    # Get input from pipeline
+    input_value = get_input()
+
+    # Convert input from dictionary to DataFrame
+    data_frame = pd.DataFrame(input_value)
+
+    # Verify it has required columns
+    cols = list(data_frame.columns)
+    REQUIRED_COLS = [
+        "employeeid",
+        "Capacity",
+        "BudgetedBillable",
+        "locationid",
+        "PracticeAreaID",
+        "BusinessUnit",
+        "contact_name",
+        "whencreated",
+    ]
+
+    if not all(col in cols for col in REQUIRED_COLS):
+        logger.error(
+            f"CSV is missing REQUIRED_COLS. Found={json.dumps(cols)}, Required={json.dumps(REQUIRED_COLS)}"
+        )
+        sys.exit(1)
```

Comment:
Consider changing to `raise` an exception like we do elsewhere when encountering problems.